frontend/src/components/NavBar.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react'
+import {
+  Button,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem,
+  Nav,
+  Navbar,
+  NavbarBrand,
+  NavItem,
+  UncontrolledDropdown
+} from 'reactstrap'
+import LanguageSelector from './LanguageSelector'
+import {withRouter} from 'react-router-dom';
+import k_logo from "../media/greenK.png"
+import kiva_logo from '../media/kivaPlainLogo.png'
+import info_image from '../media/gray_info.png'
+import sandwich_image from '../media/sandwich.png'
+import '../styles/navbar.css'
+
+const NavBarComponent = withRouter(props => <NavBar {...props}/>);
+
+class NavBar extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isLoginPage: true
+    }
+
+    this.handlePageChange = this.handlePageChange.bind(this)
+  }
+
+  handlePageChange() {
+    if (this.props.location.pathname.equals("http://localhost:3000/dashboard")) {
+      this.setState({isLoginPage: false})
+    } else {
+      this.setState({isLoginPage: true})
+    }
+  }
```

Comment:
This is on the right track - you'll probably want to use the component lifecycle function componentDidMount, which is called before the component is rendered and will allow you to check the path of a component immediately

Reply:
Also using componentDidMount means that the bind in the constructor isn't necessary as well, remember to get rid of that